torchaudio/datasets/cmudict.py

```diff
@@ -0,0 +1,170 @@
+    "(IN-PARENTHESES",
+    "(LEFT-PAREN",
+    "(OPEN-PARENTHESES",
+    "(PAREN",
+    "(PARENS",
+    "(PARENTHESES",
+    ")CLOSE-PAREN",
+    ")CLOSE-PARENTHESES",
+    ")END-PAREN",
+    ")END-PARENS",
+    ")END-PARENTHESES",
+    ")END-THE-PAREN",
+    ")PAREN",
+    ")PARENS",
+    ")RIGHT-PAREN",
+    ")UN-PARENTHESES",
+    "+PLUS",
+    ",COMMA",
+    "--DASH",
+    "-DASH",
+    "-HYPHEN",
+    "...ELLIPSIS",
+    ".DECIMAL",
+    ".DOT",
+    ".FULL-STOP",
+    ".PERIOD",
+    ".POINT",
+    "/SLASH",
+    ":COLON",
+    ";SEMI-COLON",
+    ";SEMI-COLON(1)",
+    "?QUESTION-MARK",
+    "{BRACE",
+    "{LEFT-BRACE",
+    "{OPEN-BRACE",
+    "}CLOSE-BRACE",
+    "}RIGHT-BRACE",
+])
+
+
+class CMUDict(Dataset):
+    """Create a Dataset for CMU Pronouncing Dictionary (CMUDict).
+
+    Args:
+        root (str or Path): Path to the directory where the dataset is found or downloaded.
+        url (str, optional):
+            The URL to download the dictionary from.
+            (default: ``"http://svn.code.sf.net/p/cmusphinx/code/trunk/cmudict/cmudict-0.7b"``)
+        url_symbols (str, optional):
+            The URL to download the list of symbols from.
+            (default: ``"http://svn.code.sf.net/p/cmusphinx/code/trunk/cmudict/cmudict-0.7b.symbols"``)
+        download (bool, optional):
+            Whether to download the dataset if it is not found at root path. (default: ``False``).
+    """
+
+    def __init__(self,
+                 root: Union[str, Path],
+                 url: str = "http://svn.code.sf.net/p/cmusphinx/code/trunk/cmudict/cmudict-0.7b",
+                 url_symbols: str = "http://svn.code.sf.net/p/cmusphinx/code/trunk/cmudict/cmudict-0.7b.symbols",
+                 download: bool = False,
+                 exclude_punctuations: bool = True) -> None:
+
+        self.exclude_punctuations = exclude_punctuations
+
+        root = Path(root)
+        if not os.path.isdir(root):
+            os.mkdir(root)
+
+        if download:
+            if os.path.isdir(root):
+                checksum = _CHECKSUMS.get(url, None)
+                download_url(url, root, hash_value=checksum, hash_type="md5")
+                checksum = _CHECKSUMS.get(url_symbols, None)
+                download_url(url_symbols, root, hash_value=checksum, hash_type="md5")
+            else:
+                RuntimeError(f"The argument `root` must be a path to directory, "
+                             "but '{root}' is passed in instead.")
+
+        self._root_path = root
+        basename = os.path.basename(url)
+        basename_symbols = os.path.basename(url_symbols)
+
+        with open(os.path.join(self._root_path, basename_symbols), "r") as text:
+            self._symbols = [line.strip() for line in text.readlines()]
+
+        with open(os.path.join(self._root_path, basename), "r") as text:
+            self._dictionary = self._parse_dictionary(text.readlines())
+
+    def _parse_dictionary(self, lines: List[str]):
```

Comment:
nit: since this function uses `self` only once, can we move this to a plain function?
Also, 

```python

def _parse_fidtionary(src: Iterable[str], exclude_punctuations: bool) -> List[str]:
    ...
```

Reply:
Thanks for the suggestion, this is fixed in [this commit](https://github.com/pytorch/audio/pull/1627/commits/4fa648fe192a9dd178e427d6730cebadb9d8db4f).